packages/audiofileplayer/android/src/main/java/com/google/flutter/plugins/audiofileplayer/AudiofileplayerPlugin.java

```diff
@@ -267,8 +342,8 @@ private void onLoad(MethodCall call, Result result) {
     try {
       if (call.argument(FLUTTER_PATH) != null) {
         String flutterPath = call.argument(FLUTTER_PATH).toString();
-        AssetManager assetManager = registrar.context().getAssets();
-        String key = registrar.lookupKeyForAsset(flutterPath);
+        AssetManager assetManager = context.getAssets();
+        String key = FlutterLoader.getInstance().getLookupKeyForAsset(flutterPath);
```

Comment:
Is this the canonical way to lookup assets in v2?

Would it still work for v1? We need to make these plugins compatible with both embeddings.

Reply:
Yes this works on v1. Not sure if this is the canonical way for v2 though.

Reply:
Can you elaborate on what 'v1' andd 'v2' means? Why would a plugin support both? (Is it an SDK feature only on newer SDKs, etc?)

Reply:
Android embedding is a Android platform-side wrapper that host flutter runtime. The v1(version 1) embedding is now deprecated and v2 effectively refactors, reorganizes and documents the existing APIs. It introduces a more intuitive and controllable lifecycle at both Android and Flutter engine level and that makes it easier for apps to build more advanced features such as background execution. Detailed documentation can be found here: https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects